packages/x-data-grid/src/components/panel/filterPanel/GridFilterInputBoolean.tsx

```diff
@@ -54,13 +54,14 @@ function GridFilterInputBoolean(props: GridFilterInputBooleanProps) {
     (event: React.ChangeEvent<HTMLInputElement>) => {
       const value = event.target.value;
       setFilterValueState(value);
-      applyValue({ ...item, value: Boolean(value) });
+
+      applyValue({ ...item, value: value ? String(value).toLowerCase() === 'true' : value });
```

Comment:
@arminmeh  do you think we should also support uppercase values ?

```
 filterModel: {
              items: [
                {
                  field: 'isAdmin',
                  operator: 'is',
                  value: "TRUE",
                },
              ],
            },
   ```
    
 that's why I have used `.toLowerCase()`

Reply:
Thanks for this. I have expanded this even further

Since the earlier change strives to have boolean values in the filter model, I have refactored things to be boolean first and used a helper to make sure filter still responds correctly if strings (even invalid ones) are passed via `setFilterModel` API.

I have also expanded tests to cover some of these cases.

Let me know if you see any problems with it.

@mui/xgrid 
Can someone else review this, since now I have my code in here as well

Reply:
Thanks this looks cleaner and more organized.